pkg/util/deployment/deployment.go

```diff
@@ -53,9 +55,12 @@ func GetOldReplicaSets(deployment extensions.Deployment, c clientset.Interface)
 		})
 }
 
+type rsListFunc func(string, api.ListOptions) ([]extensions.ReplicaSet, error)
```

Comment:
Isn't using `List` of rsStore and podStore the whole point of having those cache stores?